nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py

```diff
@@ -0,0 +1,237 @@
+        assert self.total_virtual_tokens > 0, "There should be at least one virtual prompt token"
+        assert (
+            self.total_virtual_tokens < max_seq_length
+        ), "virtual prompt tokens should not exceed max sequence length"
+
+        logging.info("Loading and tokenizing dataset ... ")
+
+        # Datasets is just a list of json dicts
+        if isinstance(datasets[0], dict):
+            self.load_data(datasets)
+
+        # Datasets are a list of file path strings to .json or .jsonl files
+        elif isinstance(datasets[0], str):
+            for path in datasets:
+                dataset = open(path, 'r', encoding='utf-8')
+                self.load_data(dataset)
+        else:
+            raise ValueError("Datasets must be a list of dicts or a list of filepath strings")
+
+    def load_data(self, dataset):
+        """
+        Loads a dataset by filling in the task templates specified in the config file
+        with the information from each training/inference example. Converts all input 
+        text into token ids. Also replaces the <|VIRTUAL_PROMPT_#|> placeholders in 
+        the task templates with the actual virtual prompt token ids. 
+
+        params:
+            dataset: A list of json objects or a dictionary objects each
+                     containing the information needed for a training example
+        """
+        skipped = 0
+
+        for json_line in tqdm(dataset):
+
+            # Read example dict or load the information for a single example from .json file
+            if type(json_line) == dict:
+                doc = json_line
+            else:
+                doc = json.loads(json_line)
+
+            taskname = doc["taskname"]
+            prompt_template = self.task_templates[taskname]["prompt_template"]
+            prompt_template_fields = self.task_templates[taskname]["prompt_template_fields"]
+            prompt_token_splits = self.task_templates[taskname]["prompt_token_splits"]
+            input_example = prompt_template
+
+            assert (
+                sum(prompt_token_splits) == self.total_virtual_tokens
+            ), "Sum of prompt token split values must equal total number of prompt tokens"
+            assert prompt_template.count('<|VIRTUAL_PROMPT_') == len(
+                prompt_token_splits
+            ), "The number of '<|VIRTUAL_PROMPT_n|>' markers and the number of prompt token splits must match"
+
+            # Format the input example according to the template
+            for field in prompt_template_fields:
+                if field in doc.keys():
+                    field_text = doc[field]
+                    input_example = input_example.replace('{' + field + '}', field_text)
+
+                # If some fields from the template aren't present, e.g. {answer} during inference
+                # just remove that field from the template, leaving the space blank
```

Comment:
This is a good to assume `answer` is the label. Maybe show some error if the user accidently enter the wrong `key` word.